tasks/drupal/deploy.yml

```diff
@@ -0,0 +1,164 @@
+---
+
+### Build
+
+- name: Create project folder structure Drupal
+  file:
+    path: "{{ steamengine_drupal_persistent_path }}/{{ item.key }}"
+    state: directory
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: 0770
+  with_dict: "{{ steamengine_drupal_persistent_directory }}"
+
+- name: Download build {{ steamengine_build_url }}
+  get_url:
+    url: "{{ steamengine_build_url }}"
+    dest: "{{ steamengine_project_root_path }}/project.zip"
+    checksum: "{{ steamengine_build_checksum }}"
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: u=rwx,g=rx,o=
+    headers: "{{ steamengine_build_url_headers }}"
+    validate_certs: "{{ steamengine_build_url_validate_certs }}"
+  notify:
+    - "{{ steamengine_project_name }} fixpermissions"
+    - "{{ steamengine_project_name }} restart"
+  register: get_url_build
+  tags:
+    - steamengine_deploy_drupal
+
+- name: "Is there a new build: {{ get_url_build.changed }}"
+  set_fact:
+    new_build_to_deploy: "{{ get_url_build.changed|bool }}"
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Create tmp directory
+  tempfile:
+    state: directory
+    suffix: "{{ steamengine_project_name }}"
+  register: tempdir
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_sf
+
+- name: Unarchive build
+  unarchive:
+    src: "{{ steamengine_project_root_path }}/project.zip"
+    dest: "{{ tempdir.path }}"
+    remote_src: true
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_sf
+
+- name: Ensure settings
+  copy:
+    src: "{{ tempdir.path }}/web/sites/default/default.settings.php"
+    dest: "{{ steamengine_conf_path }}/settings.php"
+    force: no
+    remote_src: true
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: u=rwx,g=rx,o=
+  register: drupal_settings
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Customize drupal settings
+  blockinfile:
+    dest: "{{ steamengine_conf_path }}/settings.php"
+    block: "{{ steamengine_drupal_project_setting_block }}"
+    mode: u=rwx,g=rx,o=
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Clean www directory
```

Comment:
It's cleaned and the the mv comment move tempdir.path to the www, thus when running with become: true, permissions are broken and www directory owner is root.

All this part must be redone, usage of "command" is a bad pratice. See how its done in other project type. You can clean the www directory and extract the zip at the right place.

Reply:
Still relevant

Reply:
C'est le fixpermissions qui s'en occupe. La solution serait de changer les droits récursivement sur l'ensemble des fichiers, c'est ce que fait le fixpermission

J'ai regardé sur les autres target, y a pas plus de truc, a part le droit sur le dosier racine, mais ça règle pas le problème des sous fichiers

Reply:
Si nginx y accêde alors que tout n'est pas près, ça va générer des fichiers foireux et péter le déployement